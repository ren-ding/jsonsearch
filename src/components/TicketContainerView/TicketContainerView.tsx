import React, { Component } from 'react';
import TicketContainer from '../../model/TicketContainer';
import ItemView from '../ItemView';

export interface TicketContainerViewProps {
    ticketContainerList: Array<TicketContainer>
}

export default class TicketContainerView extends Component<TicketContainerViewProps> {
    renderList = () => this.props.ticketContainerList.map((ticketContainer: TicketContainer, index:number) => 
      (
        <div key={index}>
          <h2>Ticket:</h2>
            <ItemView
              itemObject = {ticketContainer.ticket}
            />
          <h2>Submitter:</h2>
            <ItemView
              itemObject = {ticketContainer.submitter || {}}
            />
          <h2>Assignee:</h2>
            <ItemView
              itemObject = {ticketContainer.assignee || {}}
            />
          <h2>Organization:</h2>
            <ItemView
              itemObject = {ticketContainer.organization || {}}
            />
        </div>
      )
    );

    render() {
      return <div>{this.renderList()}</div>;
    }
}