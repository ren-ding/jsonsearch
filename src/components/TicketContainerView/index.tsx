import TicketContainerView from './TicketContainerView';
export default TicketContainerView;