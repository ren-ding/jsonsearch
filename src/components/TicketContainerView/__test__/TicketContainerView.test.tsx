import React from 'react';
import TicketContainerView, {TicketContainerViewProps} from '../TicketContainerView';
import renderer from 'react-test-renderer';

describe('TicketContainerView', () => {
    let props:TicketContainerViewProps;
    beforeEach(()=> {
        props = {
            ticketContainer: {
                ticket: {
                    "_id": "436bf9b0-1147-4c0a-8439-6f79833bff5b",
                    "url": "http://initech.zendesk.com/api/v2/tickets/436bf9b0-1147-4c0a-8439-6f79833bff5b.json",
                    "external_id": "9210cdc9-4bee-485f-a078-35396cd74063",
                    "created_at": "2016-04-28T11:19:34 -10:00",
                    "type": "incident",
                    "subject": "A Catastrophe in Korea (North)",
                    "description": "Nostrud ad sit velit cupidatat laboris ipsum nisi amet laboris ex exercitation amet et proident. Ipsum fugiat aute dolore tempor nostrud velit ipsum.",
                    "priority": "high",
                    "status": "pending",
                    "submitter_id": 1,
                    "tags": [
                      "Ohio",
                      "Pennsylvania",
                      "American Samoa",
                      "Northern Mariana Islands"
                    ],
                    "has_incidents": false,
                    "due_at": "2016-07-31T02:37:50 -10:00",
                    "via": "web"
                  },
                submitter: {
                    "_id": 1,
                    "url": "http://initech.zendesk.com/api/v2/users/1.json",
                    "external_id": "74341f74-9c79-49d5-9611-87ef9b6eb75f",
                    "name": "Francisca Rasmussen",
                    "alias": "Miss Coffey",
                    "created_at": "2016-04-15T05:19:46 -10:00",
                    "active": true,
                    "verified": true,
                    "shared": false,
                    "locale": "en-AU",
                    "timezone": "Sri Lanka",
                    "last_login_at": "2013-08-04T01:03:27 -10:00",
                    "email": "coffeyrasmussen@flotonic.com",
                    "phone": "8335-422-718",
                    "signature": "Don't Worry Be Happy!",
                    "organization_id": 119,
                    "tags": [
                    "Springville",
                    "Sutton",
                    "Hartsville/Hartley",
                    "Diaperville"
                    ],
                    "suspended": true,
                    "role": "admin"
                },
                assignee: undefined,
                organization: undefined,
            }
        }
    });

    it('should render to match snapshot', () => {
        const tree = renderer.create(<TicketContainerView {...props}/>).toJSON();
        expect(tree).toMatchSnapshot();
    });
});