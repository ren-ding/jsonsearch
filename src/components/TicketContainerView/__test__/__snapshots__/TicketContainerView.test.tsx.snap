// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TicketContainerView should render to match snapshot 1`] = `
<div>
  <div>
    <h2>
      Ticket:
    </h2>
    <div>
      <table
        className="ui compact table"
      >
        <thead>
          <tr>
            <th>
              Field
            </th>
            <th>
              Value
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              _id
            </td>
            <td>
              436bf9b0-1147-4c0a-8439-6f79833bff5b
            </td>
          </tr>
          <tr>
            <td>
              url
            </td>
            <td>
              http://initech.zendesk.com/api/v2/tickets/436bf9b0-1147-4c0a-8439-6f79833bff5b.json
            </td>
          </tr>
          <tr>
            <td>
              external_id
            </td>
            <td>
              9210cdc9-4bee-485f-a078-35396cd74063
            </td>
          </tr>
          <tr>
            <td>
              created_at
            </td>
            <td>
              2016-04-28T11:19:34 -10:00
            </td>
          </tr>
          <tr>
            <td>
              type
            </td>
            <td>
              incident
            </td>
          </tr>
          <tr>
            <td>
              subject
            </td>
            <td>
              A Catastrophe in Korea (North)
            </td>
          </tr>
          <tr>
            <td>
              description
            </td>
            <td>
              Nostrud ad sit velit cupidatat laboris ipsum nisi amet laboris ex exercitation amet et proident. Ipsum fugiat aute dolore tempor nostrud velit ipsum.
            </td>
          </tr>
          <tr>
            <td>
              priority
            </td>
            <td>
              high
            </td>
          </tr>
          <tr>
            <td>
              status
            </td>
            <td>
              pending
            </td>
          </tr>
          <tr>
            <td>
              submitter_id
            </td>
            <td>
              1
            </td>
          </tr>
          <tr>
            <td>
              tags
            </td>
            <td>
              Ohio,Pennsylvania,American Samoa,Northern Mariana Islands
            </td>
          </tr>
          <tr>
            <td>
              has_incidents
            </td>
            <td>
              false
            </td>
          </tr>
          <tr>
            <td>
              due_at
            </td>
            <td>
              2016-07-31T02:37:50 -10:00
            </td>
          </tr>
          <tr>
            <td>
              via
            </td>
            <td>
              web
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <h2>
      Submitter:
    </h2>
    <div>
      <table
        className="ui compact table"
      >
        <thead>
          <tr>
            <th>
              Field
            </th>
            <th>
              Value
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              _id
            </td>
            <td>
              1
            </td>
          </tr>
          <tr>
            <td>
              url
            </td>
            <td>
              http://initech.zendesk.com/api/v2/users/1.json
            </td>
          </tr>
          <tr>
            <td>
              external_id
            </td>
            <td>
              74341f74-9c79-49d5-9611-87ef9b6eb75f
            </td>
          </tr>
          <tr>
            <td>
              name
            </td>
            <td>
              Francisca Rasmussen
            </td>
          </tr>
          <tr>
            <td>
              alias
            </td>
            <td>
              Miss Coffey
            </td>
          </tr>
          <tr>
            <td>
              created_at
            </td>
            <td>
              2016-04-15T05:19:46 -10:00
            </td>
          </tr>
          <tr>
            <td>
              active
            </td>
            <td>
              true
            </td>
          </tr>
          <tr>
            <td>
              verified
            </td>
            <td>
              true
            </td>
          </tr>
          <tr>
            <td>
              shared
            </td>
            <td>
              false
            </td>
          </tr>
          <tr>
            <td>
              locale
            </td>
            <td>
              en-AU
            </td>
          </tr>
          <tr>
            <td>
              timezone
            </td>
            <td>
              Sri Lanka
            </td>
          </tr>
          <tr>
            <td>
              last_login_at
            </td>
            <td>
              2013-08-04T01:03:27 -10:00
            </td>
          </tr>
          <tr>
            <td>
              email
            </td>
            <td>
              coffeyrasmussen@flotonic.com
            </td>
          </tr>
          <tr>
            <td>
              phone
            </td>
            <td>
              8335-422-718
            </td>
          </tr>
          <tr>
            <td>
              signature
            </td>
            <td>
              Don't Worry Be Happy!
            </td>
          </tr>
          <tr>
            <td>
              organization_id
            </td>
            <td>
              119
            </td>
          </tr>
          <tr>
            <td>
              tags
            </td>
            <td>
              Springville,Sutton,Hartsville/Hartley,Diaperville
            </td>
          </tr>
          <tr>
            <td>
              suspended
            </td>
            <td>
              true
            </td>
          </tr>
          <tr>
            <td>
              role
            </td>
            <td>
              admin
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <h2>
      Assignee:
    </h2>
    <div>
      <table
        className="ui compact table"
      >
        <thead>
          <tr>
            <th>
              Field
            </th>
            <th>
              Value
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
    <h2>
      Organization:
    </h2>
    <div>
      <table
        className="ui compact table"
      >
        <thead>
          <tr>
            <th>
              Field
            </th>
            <th>
              Value
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
  </div>
</div>
`;
